packages/wmr/src/lib/prerender.js

```diff
@@ -155,19 +166,27 @@ async function workerCode({ cwd, out, publicPath }) {
 				routes.push({ url, _discoveredBy: route });
 			}
 		}
-		const body = (result && result.html) || result;
+
+		let body;
+		if (result && typeof result === 'object') {
+			if (result.html) body = result.html;
+			if (result.head) {
+				head = result.head;
+			}
+		} else {
+			body = result;
+		}
+
+		// TODO: Use a proper HTML parser here. We should definitely not parse HTML
+		// with regex :S
 
 		// Inject HTML links at the end of <head> for any stylesheets injected during rendering of the page:
-		let headHtml = [
-			document.head[BEFORE],
-			...new Set(head.filter(c => c.rel && c.href).map(c => `<link rel="${enc(c.rel)}" href="${enc(c.href)}">`)),
-			document.head[AFTER]
-		].join('');
+		let headHtml = head.elements ? Array.from(serializeHead(head.elements)).join('') : '';
```

Comment:
Adding support for recursion (2/2)

```suggestion
		let headHtml = head.elements && serializeElement(head.elements);
```